primitives/staking/src/lib.rs

```diff
@@ -54,67 +56,93 @@ impl<AccountId, Balance> OnStakerSlash<AccountId, Balance> for () {
 	}
 }
 
-/// Trait for communication with the staking pallet.
+/// A struct that reflects stake that an account has in the staking system. Provides a set of
+/// methods to operate on it's properties. Aimed at making `StakingInterface` more concise.
+pub struct Stake<T: StakingInterface + ?Sized> {
```

Comment:
In the staking pallet, a

`impl From<StakingLedger> for sp_staking::Stake` 

might come handy now.

Reply:
Can be done, though it's only going to be used in one method in staking pallet, so not sure if this is too helpful or more DRY than it is now